### ISSUE TITLE
'$this' can not be used in static closures on arrow function

### ISSUE THREAD

**Issue by jorgsowa** [kalessil/phpinspectionsea#1851]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) |
| Language level | PHP 8.0                        |

Current behaviour
--
Similar issue to [the one fixed before](https://github.com/kalessil/phpinspectionsea/issues/1194), but this case is about static arrow function which is not covered.

```PHP
<?php declare(strict_types=1);

final class Foo
{
    public function bar(): Closure
    {
        return static fn() => $this->haha();
    }

    private function haha(): string
    {
        return 'I love Inspections!';
    }
}

(new Foo())->bar()();
```

This code result in error `PHP Fatal error:  Uncaught Error: Using $this when not in object context`, but it doesn't show any warning.

Expected behaviour
--

Display warning `'$this' can not be used in static closures`

Kind regards!

**Comment by janikaja-tet** [comment]
Guys, please, implement this feature! I just encountered this issue and could not understand for a while what was wrong 😕
